extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1013,6 +1054,101 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return ['https://partner.googleadservices.com',
       'https://tpc.googlesyndication.com'];
   }
+
+  /** @override */
+  getNonAmpCreativeRenderingMethod(headerValue) {
+    if (this.isFluid_) {
+      return XORIGIN_MODE.SAFEFRAME;
+    }
+    return super.getNonAmpCreativeRenderingMethod(headerValue);
+  }
+
+  /** @override */
+  getAdditionalContextMetadata() {
+    const attributes = /** @type {!JsonObject} */ ({});
+    if (this.isFluid_) {
+      attributes['uid'] = 1;
+      attributes['hostPeerName'] = this.win.location.origin;
+      // The initial geometry isn't used for anything important, but it is
+      // expected, so we pass this string with all zero values.
+      attributes['initialGeometry'] = JSON.stringify(
```

Comment:
Are we sure we need to stringify this?  I thought the whole attributes object gets stringified

Reply:
Yeah, we need this. Safeframe expects these to be strings on its end.